Xmltohb latest

deepakala-k opened the pull request.

deepakala-k committed:
Add hw access device paths at the MRW processing stage

For scom and cfam access, we need the backend device paths for the specific
chip. Adding the device paths based on a map corresponding to respective proc
or ocmb chips.

Tested and the generated xml contains the device paths as per the chip.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Use device paths populated at the MRW processed stage

Update the filter to include the device paths, so that it gets added into the device tree.

Tested and works as expected.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Moved the existing files from hostboot

deepakala-k committed:
Update Makefile to include targeting changes

Adjust the Makefile to incorporate recent targeting updates, ensuring
that the build process includes the new filtering logic and generated
files required for targeting.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add support for attribute- and target-level filtering

In addition to the existing filtering based on the target_types file,
we now support two new input lists:
 - an attribute filter list to select specific attributes within each target
 - a target filter list to restrict which targets are included

These lists are applied during processing to provide finer-grained
filtering control.

--filter-attr-file and --filter-tgt-file flag enables that support

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Fix duplicate XML declarations and <attributes> tags when merging

Merging multiple XML files was introducing duplicate XML declarations
as well as repeated <attributes> and </attributes> tags, which caused
parsing issues. This change removes all intermediate declarations and
attribute tags, enclosing the final merged output within a single
<attributes> block. This provides a generic and consistent solution.

And if the input files does not exist or is empty throw error

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Rename command options from "hb" to "plat"

Update the command-line options to replace the term "hb" with "plat",
which is a more common and appropriate name. This makes the option
naming consistent and usable across all contexts.

If there is a targetTypeExtension that does not have a targetType, give out a
warning

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add CLI options for attribute/target filtering and BMC build flag

Introduce new command-line options to:
 - enable attribute filtering
 - enable target filtering
 - indicate if a BMC build is in progress

When filtering files are provided, header generation now includes only
the required attribute information.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Use BMC build flag to adjust header includes

When the BMC build flag is enabled, the generated header files now
reference a different set of include files. This ensures the correct
headers are used for BMC builds.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Introduce attribute and target filter file definitions

Add two new files that define the lists of attributes and targets to be
used for filtering during build processing.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add the targettype file required to support OBMC

Include the targetTypeExtension that adds the additional attributes needed for
the specific targets and the extra targetType that are exculsively needed only
for OBMC

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add Customize ekb attrs file

This commit contains a file which defines the temporary FW defaults for ekb
(FAPI) attributes needing a default. Each FAPI attribute should already be
defined elsewhere in an attribute XML file.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add attribute definition for BMC specific attributes

Add the complete attribute definition for the attributes that are added only to
support OBMC.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add required EKB attributes and target xml list

Limit to only the EKB files used by HWPs invoked from the BMC for
generating attributes and target definitions. From these input files,
`attribute_type_ekb` and `target_type_ekb` are produced.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Dummy files needed for running the step in xml creation

No changes are included in this commit.
This adds a placeholder in attribute_service.H for future
work that will introduce attribute override handling and
methods to access the targeting service. The upcoming
implementation will provide GET and SET macros for attributes.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add bash script for generating targeting final XML and headers

This script automates the full flow of target and attribute generation:
 - merges attributes and targets
 - applies filtering
 - creates EKB targets and attributes
 - processes MRW files and merges them into the output XML
 - generates the header files required to build the targeting library

The script encapsulates these multiple steps into a single process.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Fix missing AFFINITY_PATH override for target

Each target instance is required to override the AFFINITY_PATH
attribute; otherwise, the build fails. This commit updates the missing
AFFINITY_PATH for one of the targets to comply with the rule.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Update AttrIdNameMap to include AFFINITY_PATH and native type attributes

As required by the new targeting code, AFFINITY_PATH must now be present
in AttrIdNameMap. This commit also adds the native type attributes to
ensure the map is complete.

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
getTargetAttributes: apply filter list to include only required attributes

Update getTargetAttributes to use the provided filter list so that only
the attributes explicitly required are included in the output. This
reduces unnecessary data and aligns with the filtering mechanism used

Signed-off-by: Deepa Karthikeyan <deepakala.karthikeyan@ibm.com>

devenrao commented on targeting/common/xmltohb/xmltohb.pl:
```diff
@@ -5932,6 +5932,24 @@ sub mergeComplexAttributeFields {
     return $mergedFields;
 }
 
+sub loadAllowedFilterList {
+    my ($filter_file, $allowed_list_ref, $filter_is_active_ref) = @_;
```
fix commit meesage heading size is more than >80

deepakala-k replied:
done. Thank you

deepakala-k committed:
Apply formatting

SwethaParasa commented on targeting/xmltohb/p10/filter_AttributesList.lsv:
```diff
@@ -0,0 +1,115 @@
+CLOCK_MUX0C_RCS_PLL_INPUT
+CLOCK_MUX0D_RCS_PLL_INPUT
+CLOCK_MUX10_PAU_DPLL_INPUT
+CLOCK_MUX11_NEST_DPLL_INPUT
```
What is the reason behind removing FA and CA prefixes from this file? When we need to find which type of attribute it is for use case like adding a default value etc., it could be easier to find if it is a FA or CA and make changes in either bmc_customized_ekb_attrs.xml or target_types_obmc.xml

deepakala-k replied:
The FA and CA prefixes were not completely correct, some of the Fapi attributes were marked as CA. It was mainly that they expected the FA to have ATTR prefix, so while filtering ATTR prefix was needed. But in targeting, handle_fapi_attr_mapping.pl will take care of using the using the <attribute><ID> mapping to the <hwpfToHbAttrMap><id> 

So we can filter using just the ID itself. So removed this special code.

 <attribute>
    <id>EC</id>
    <description>attribute indicating the chip target's EC level</description>
    <hwpfToHbAttrMap>
      <id>ATTR_EC</id>
    </hwpfToHbAttrMap>
  </attribute>

To find if the attribute is a FAPI attribute, we can look in the attribute_types_ekb file.

gcpin commented on targeting/common/xmltohb/xmltohb.pl:
```diff
@@ -0,0 +1,9606 @@
+//******************************************************************************
+
+// STD
+#include <builtins.h>
```
As discussed, please change this to <targeting/xmltohb/builtins.h>

deepakala-k replied:
I will push the updated code shortly

gcpin commented on targeting/common/xmltohb/xmltohb.pl:
```diff
@@ -0,0 +1,9606 @@
+#include <stdlib.h>
+
+// Targeting component
+#include <builtins.h>
```
Also here please change to <targeting/xmltohb/builtins.h>

deepakala-k replied:
Sure. Done

deepakala-k committed:
Add PARENT_PERVASIVE attribute for FC targets

During p9 days, pervasive parent was explicitly added for each unit. But it was
moved up and generalized. So this hack of not adding parent_pervasive attribute
for FC targets can be removed.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Remove FSI target for bmc

The fsi path details are added as attributes to the chip targets. The code that
is removed as part of this commit, used to add the FSI details to a separate
target. That approach makes it diffcult to dynamically switch backends.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Migrate filter list from LSV to XML

Earlier, LSV files were used for defining the filter list. This has now
been moved to XML files to maintain uniformity across components. Using
XML also provides flexibility to add additional tags in the future to
describe more properties for filtering.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Update scripts to use new XML filter files

Updated all relevant scripts to perform filtering based on the new XML
filter files instead of the old LSV files.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add utility to inherit attributes from parent target types

Created a script to add attributes of a parent target type
to its child target type. The parent target is defined using a
<parent>id</parent> tag, and the script recursively traverses through
all levels of parent relationships to propagate attributes to the
children.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add DTS generation script using final XML input

Added a script to generate Device Tree Source (DTS) files from the final
XML, which combines target instances and target types. The script picks
attribute values in the following order of priority:
1. Target instance value
2. Target type default
3. Attribute default

The DTS is generated following the physical path hierarchy. Both the
PHYS_PATH and AFFINITY_PATH are converted into their respective binary
formats for accurate device tree representation.

Values are packed based on device tree conventions:
- <> for uint8, uint16, uint32 types
- [] for arrays
- "" for strings

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Integrate DTS generation script into build flow

Updated the bash script to invoke the DTS generation script as the next
step in the build process. The DTS generation now becomes part of the
overall build sequence, ensuring that the latest XML data is always used
to produce the device tree output.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Workaround: op-build hostfw bump should fix this. Copying the dtb to image

deepakala-k committed:
switch to common Hostboot DTS generation scripts

Previously, the device tree was generated using BMC-specific scripts.
The Makefile has been updated to remove the invocation of the old
BMC-specific scripts.

Signed-off-by: deepakalak <deepakala.karthikeyan@ibm.com>

deepakala-k committed:
Add missing attribute needed for HWP

deepakala-k committed:
Create targeting dump source file

deepakala-k committed:
Create platform attributes to provide writeable property
